main/zoe/guide/contracts/simple-exchange.md

```diff
@@ -2,116 +2,135 @@
 ```
 
 ## Buying an order
 
-Let's say that Bob hears about Alice's invite and decides he wants to
-join. He will check the `installationHandle` and see that the exchange
-is trading what he
-expects:
+Let's say that Bob knows about the simple exchange, and hears about Alice's
+offer, which sounds like a good deal to him. He will check the installation
+with zoe and see that the exchange is trading what he expects:
 
 ```js
-// Bob collects information
-const inviteIssuer = zoe.getInviteIssuer();
-const bobExclusiveInvite = await inviteIssuer.claim(bobInvite);
-const { instanceHandle } = inviteIssuer.getAmountOf(bobExclusiveInvite).value[0];
+const bobInvitation = E(publicFacet).makeInvitation();
+const invitationIssuer = E(zoe).getInvitationIssuer();
+const bobExclusiveInvitation = E(invitationIssuer).claim(bobInvitation);
 const {
-  installationHandle: bobInstallationId,
-  issuerKeywordRecord: bobIssuers,
-} = zoe.getInstance(instanceHandle);
+  instance,
+  installation,
+} = await E(zoe).getInvitationDetails(bobExclusiveInvitation);
+const bobIssuers = await E(zoe).getIssuers(instance);
+```
+
+Bob verifies that the information is what he expects. He can compare the
+installation he gets from the invitation with a canonical link he found in a
+public directory he trusts.
 
-// Bob checks the information is what he expects
-assert(bobInstallationId === installationHandle, details`wrong installation`);
+```js
+assert(installation === simpleExchangeInstallation, details`wrong installation`);
 assert(bobIssuers.Asset === moolaIssuer, details`wrong Asset issuer`);
 assert(bobIssuers.Price === simoleanIssuer, details`wrong Price issuer`);
 ```
 
 Now that Bob has checked to make sure everything is in order, he proceeds to fulfill the buy order:
 
 ```js
-// Bob creates his proposal
 const bobBuyOrderProposal = harden({
-  give: { Price: simoleans(7) },
-  want: { Asset: moola(3) },
+  give: { Price: simoleanAmountMath(7) },
+  want: { Asset: moolaAmountMath.make(3) },
   exit: { onDemand: null },
 });
 
-const bobPayments = { Price: bobSimoleanPayment };
+const bobSimPayment = await E(bobSimoleanPurse).withdraw(simoleanAmountMath(7));
+onst bobPayments = { Price: bobSimPayment };
 
-// Bob escrows with zoe and adds an order to the exchange
-const { outcome: bobOfferResult, payout: bobPayoutP } = await E(zoe).offer(
-  bobExclusiveInvite,
+const bobSeat = await E(zoe).offer(
+  bobExclusiveInvitation,
   bobBuyOrderProposal,
   bobPayments,
 );
-
 ```
 
 ## Payout
 
-If a match is made, the payout promise that the user receives when
-they escrow with Zoe is resolved to a record of promises for payments with keyword keys:
+When a match is made, the payout promise that the user can get from their seat
+is resolved to a promises for payment.
 
 ```js
-const bobPayout = await bobPayoutP;
-const alicePayout = await alicePayoutP;
+const { Asset: bobAssetPayout, Price: bobPricePayout } = await bobSeat.getPayouts();
```

Comment:
This won't work as written, I think because `bobAssetPayout` will be a promise still and we can't deposit those.

Reply:
fixed.